src/SGFTree.cpp

```diff
@@ -124,7 +124,9 @@ void SGFTree::populate_states(void) {
             if (!m_properties.count("SZ")) {
                 // No size, but SGF spec defines default size for Go
                 m_properties.insert(std::make_pair("SZ", "19"));
-                valid_size = true;
+                if (19 == BOARD_SIZE) {
+                    valid_size = true;
```

Comment:
valid_size only says whether m_boardsize has been set, not whether we can deal with it or not. So I think you don't need changes here. Later on, you use valid_size to see what the board has been set at, and there's a check that it was either set or defaulted to BOARD_SIZE.